velox/exec/QueryConfigTracer.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/exec/QueryConfigTracer.h"
+#include "velox/exec/TraceWriter.h"
+
+namespace facebook::velox::exec {
+namespace {
+const std::string kQueryConfigKey = "queryConfig";
+const std::string kConnectorPropertiesKey = "connectorProperties";
+} // namespace
+
+QueryConfigTracer::QueryConfigTracer(std::string path)
+    : pool_(memory::MemoryManager::getInstance()
+                ->testingSharedLeafPools()[0]
+                .get()),
+      path_(std::move(path)) {
+  const auto fs = filesystems::getFileSystem(path_, nullptr);
+  if (!fs->exists(path_)) {
+    fs->mkdir(path_);
+  }
+  configFile_ = fmt::format("{}/query_config.json", path_);
+}
+
+void QueryConfigTracer::write(
+    const std::shared_ptr<core::QueryCtx>& queryCtx) const {
+  folly::dynamic querConfigObj = folly::dynamic::object;
+  for (const auto& [key, value] : queryCtx->queryConfig().values()) {
+    querConfigObj[key] = value;
+  }
+
+  folly::dynamic connectorPropertiesObj = folly::dynamic::object;
+  for (const auto& [connectorId, configs] :
+       queryCtx->connectorSessionProperties()) {
+    folly::dynamic obj = folly::dynamic::object;
+    for (const auto& [key, value] : configs->values()) {
+      obj[key] = value;
+    }
+    connectorPropertiesObj[connectorId] = obj;
+  }
+
+  folly::dynamic configObj = folly::dynamic::object;
+  configObj[kQueryConfigKey] = querConfigObj;
+  configObj[kConnectorPropertiesKey] = connectorPropertiesObj;
+  const auto configStr = folly::toJson(configObj);
+
+  std::ofstream configFile(configFile_);
```

Comment:
Can we use velox file system to write? Otherwise it only works for local fs. Thanks!